### ISSUE TITLE
3.3: failure running tests

### ISSUE THREAD

**Issue by sandrotosi** [txt2tags/txt2tags#211]
Hello,
i'm preparing txt2tags 3.3 for debian, and during our built process we also run tests; the test suite fails on a single test:

```
Entering module marks
  comment OK
  image OK
  inline OK
  line OK
  link OK
  list OK
  macro FAILED
meld ok/macro.html /home/morph/deb/build-area/txt2tags-3.3/test/marks/error/macro.html
--- 

+++ 

@@ -34,8 +34,8 @@

 <h2>Syntax: Macro without formatting string</h2>
 
 <p>
-Date    : @DATE@ - @DATE@<br>
-Mtime   : @MTIME@ - @MTIME@<br>
+Date    : 20191224 - 20191224<br>
+Mtime   : 20191224 - 20191224<br>
 Infile  : macro.t2t - macro.t2t<br>
 Outfile : macro.html - macro.html<br>
 </p>
@@ -45,8 +45,8 @@

 <h2>Syntax: Macro name is case insensitive</h2>
 
 <p>
-Date    : @DATE@<br>
-Mtime   : @MTIME@<br>
+Date    : 20191224<br>
+Mtime   : 20191224<br>
 Infile  : macro.t2t<br>
 Outfile : macro.html<br>
 </p>
@@ -89,8 +89,8 @@

 <h2>Test: Various per line, glued</h2>
 
 <p>
-Date    : 20@DATE@@DATE@<br>
-Mtime   : 20@MTIME@@MTIME@<br>
+Date    : 202019122420191224<br>
+Mtime   : 202019122420191224<br>
 Infile  : t2tmacro.t2tmacro.t2t<br>
 Outfile : htmlmacro.htmlmacro.html<br>
 </p>
  paragraph OK

...

Totals: 234 tests (233 ok / 1 failed)

Check out the files with errors:
....../txt2tags-3.3/test/marks/error/macro.html
```

and that file is: 
[macro.html.txt](https://github.com/txt2tags/txt2tags/files/3997235/macro.html.txt)

it's unclear to me if this is a direct effect of a patch we maintain to make the build reproducible (https://salsa.debian.org/python-team/applications/txt2tags/blob/debian/master/debian/patches/reproducible.patch) as they are both related to dates.

can you please have a look?

thanks!

**Comment by jendrikseipp** [comment]
Thanks for bringing this to our attention. I'm planning to remove the macro functionality from txt2tags, so this issue should be fixed automatically. Will keep you posted.

**Comment by jendrikseipp** [comment]
I removed the macros and released version 3.4. If the test still don't pass, please open another issue. Looking forward to seeing the new txt2tags version in Debian!

**Comment by sandrotosi** [comment]
`Totals: 227 tests (100% ok)` , thanks!

i needed to update the shebangs of txt2tags.py and test/run.py to `/usr/bin/env python3` : since we are targetting py3k only in Debian that's fine, but otherwise invoking those scripts directly (like to run tests) requires python2 (aka `python`)
